crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -461,13 +528,20 @@ where
         let fragment_required = !targets.is_empty()
             || key.mesh_key.contains(MeshPipelineKey::DEPTH_CLAMP_ORTHO)
             || (key.mesh_key.contains(MeshPipelineKey::MAY_DISCARD)
-                && self.material_fragment_shader.is_some());
+                && self.prepass_material_fragment_shader.is_some());
 
         let fragment = fragment_required.then(|| {
             // Use the fragment shader from the material
-            let frag_shader_handle = match self.material_fragment_shader.clone() {
-                Some(frag_shader_handle) => frag_shader_handle,
-                _ => PREPASS_SHADER_HANDLE,
+            let frag_shader_handle = if key.mesh_key.contains(MeshPipelineKey::DEFERRED_PREPASS) {
+                match self.deferred_material_fragment_shader.clone() {
+                    Some(frag_shader_handle) => frag_shader_handle,
+                    _ => PREPASS_SHADER_HANDLE,
```

Comment:
Is this correct? Shouldn't it be a default deferred prepass shader?

Reply:
In the `_ => PREPASS_SHADER_HANDLE,` case the prepass shader handle already supports deferred. This would be used where there is no material present. I currently have https://github.com/DGriffin91/bevy/blob/78560858fa4814bfece2069c776855645aa2ece2/crates/bevy_pbr/src/prepass/prepass.wgsl#L133 
Deferred doesn't really work in this case since a layer id would need to be set. Usually that id would be set by the material so I'm not sure what to do in this case since it's not available.

Reply:
Me neither, aside from setting it to a reserved 'invalid' id?

Reply:
If there were an invalid id nothing would render at all for that invalid id unless we setup an additional render pass just for the invalid id.